src/java_tools/buildjar/java/com/google/devtools/build/buildjar/javac/JavacOptions.java

```diff
@@ -283,8 +293,13 @@ public void normalize(List<String> normalized) {
         if (target != null) {
           normalized.add("-target");
           normalized.add(target);
+          releaseOrTarget = target;
         }
       }
+      // Put -parameters option back if no target option was provided or at least target 8.
+      if (parameters && Optional.ofNullable(releaseOrTarget).map(Ints::tryParse).orElse(8) >= 8) {
```

Comment:
This looks complicated and hard to understand/maintain. I'd suggest using int for releaseOrTarget, something like this:

```
      int targetedJavaVersion = 8;
      if (release != null) {
        normalized.add("--release");
        normalized.add(release);
        targetedJavaVersion = Integer.parseInt(release);
      } else {
        if (source != null) {
          normalized.add("-source");
          normalized.add(source);
        }
        if (target != null) {
          normalized.add("-target");
          normalized.add(target);
          targetedJavaVersion = Integer.parseInt(target);
        }
      }
      // Put -parameters option back if no target option was provided or at least target 8.
      if (parameters && targetedJavaVersion >= 8) {
        normalized.add("-parameters");
      }
```

Reply:
Done.